src/transformers/generation/utils.py

```diff
@@ -496,7 +496,7 @@ def _prepare_encoder_decoder_kwargs_for_generation(
                 add_hook_to_module(encoder, AlignDevicesHook(io_same_device=True))
 
         # 2. Prepare encoder args and encoder kwargs from model kwargs and generation config.
-        irrelevant_prefix = ["decoder_", "cross_attn", "use_cache"]
+        irrelevant_prefix = ["decoder_", "cross_attn", "use_cache", "past_key_values"]
```

Comment:
we don't want to populate cache when running the encoder, otherwise the tests will fail with shape mismatch for the decoder part